lib/DimmableLightDevice.js

```diff
@@ -1,15 +1,24 @@
 'use strict';
 
-const util = require('/node_modules/homey-meshdriver/lib/util');
+const util = require('homey-meshdriver/lib/util');
+
+const ZigBeeLightDevice = require('homey-meshdriver').ZigBeeLightDevice;
+
+class DimmableDevice extends ZigBeeLightDevice {
 
-module.exports = {
 	onFlowActionSetDimWithCustomTransitionTime(dimtype, dimwaarde, transitietijd) {
-		this.log('type_to_dim', parseInt(dimtype, 0));
+		dimtype = parseInt(dimtype, 10);
+
+		this.log('type_to_dim', dimtype);
 		this.log('dim_value', dimwaarde);
 		this.log('transition_value', transitietijd);
 
+		// Clear already running intervals
```

Comment:
If this function is called multiple times I assume the lamp will stop it's previous action which means the interval of the previous call can be cancelled if it exists. This will also prevent multiple intervals running at the same time polling for the same updates.